src/sage/categories/finite_fields.py

```diff
@@ -243,5 +245,283 @@ def _element_of_factored_order(self, F):
+                return True
+            q = self.parent().order()
+            if q % 2  == 0:
+                return True
+            character = self**((q-1)//2)
+            is_square = character == self.parent().one()
+            return is_square
+
+        def _tonelli(self):
+            """
+            Returns a square root of the element if it exists
+            using tonelli's algorithm
+
+            OUTPUT:
+
+            - A square root of the element; raises an error
+              if the element is not a square
+
+            EXAMPLES::
+                sage: k.<a> = GF((5, 10))
+                sage: k(2).is_square()
+                True
+                sage: k(2)._tonelli()^2 == k(2)
+                True
+                sage: k.quadratic_non_residue()._tonelli()
+                Traceback (most recent call last):
+                ...
+                ValueError: Element is not a square
+            """
+            q = self.parent().cardinality()
+            if not self.is_square():
+                raise ValueError("Element is not a square")
+            g = self.parent().quadratic_non_residue()
+            odd_order = (q - 1).odd_part()
+            even_exp = Integer.valuation(q-1, 2)
+            e = 0
+            for i in range(2, even_exp+1):
+                tmp = self * (pow(g, -e))
+
+                condition = tmp**((q-1)//(2**i)) != self.parent().one()
+                if condition:
+                    e = 2**(i-1) + e
+            h = self * (g**(-e))
+            b = g**(e//2) * h**((odd_order+1)//2)
+            return b
+
+        def _cipolla(self):
+            """
+            Returns a square root of the element if it exists
+            using cipolla's algorithm, more suited if order - 1
+            is highly divisible by 2
+
+            OUTPUT:
+
+            - A square root of the element; raises an error
+              if the element is not a square
+
+            EXAMPLES::
+                sage: k.<a> = GF((5, 10))
+                sage: k(2).is_square()
+                True
+                sage: k(2)._cipolla()^2 == k(2)
+                True
+                sage: k.quadratic_non_residue()._cipolla()
+                Traceback (most recent call last):
+                ...
+                ValueError: Element is not a square
+            """
+            parent = self.parent()
+            q = parent.cardinality()
+            if not self.is_square():
+                raise ValueError("Element is not a square")
+            t = parent.random_element()
+            root = t**2 - 4 * self
+            while root.is_square():
+                t = parent.random_element()
+                root = t**2 - 4 * self
+            from sage.rings.polynomial.polynomial_ring import polygen
+            X = polygen(parent)
+            f = X**2 - t*X + self
+            b = pow(X, (q+1)//2, f)
+            return b
+
+        def sqrt(self, all=False, algorithm='Tonelli'):
```

Comment:
```suggestion
        def sqrt(self, all=False, algorithm='tonelli'):
```

The algorithm parameter in Sage is typically a lowercase string.